add tutorial about zmod37

kbuzzard opened the pull request:
Reference to a mathlib file which no longer exists has been removed, and replaced by a more user-friendly example of an equivalence relation.

TO CONTRIBUTORS:

Make sure you have:

  * [x ] reviewed and applied the coding style: [coding](https://github.com/leanprover/mathlib/blob/master/docs/style.md), [naming](https://github.com/leanprover/mathlib/blob/master/docs/naming.md)
  * [ ] for tactics:
     * [ ] added or adapted documentation in [tactics.md](https://github.com/leanprover/mathlib/blob/master/docs/tactics.md)
     * [ ] write an example of use of the new feature in [tactics.lean](https://github.com/leanprover/mathlib/blob/master/tests/tactics.lean)
  * [ ] make sure definitions and lemmas are put in the right files
  * [ ] make sure definitions and lemmas are not redundant

If this PR is related to a discussion on Zulip, please include a link in the discussion.

For reviewers: [code review check list](https://github.com/leanprover/mathlib/blob/master/docs/code-review.md)

***********

linear_algebra/quotient_module.lean no longer exists; updating docs.

kbuzzard committed:
fix broken reference

Reference to a mathlib file which no longer exists has been removed, and replaced by a more user-friendly example of an equivalence relation.

robertylewis reviewed (approved):
@PatrickMassot This looks good to me.

kbuzzard committed:
Adding Lean example of equivalence relation usage

sgouezel commented on docs/tutorial/Zmod37.lean:
```diff
@@ -0,0 +1,195 @@
+theorem cong_mod_equiv : equivalence (cong_mod37) :=
+⟨cong_mod_refl,cong_mod_symm,cong_mod_trans⟩
+
+instance Z_setoid : setoid ℤ := { r := cong_mod37, iseqv := cong_mod_equiv }
```
Could you add a comment that putting a global setoid instance on Z is OK for a tutorial, but definitely a bad idea in the wild, and that one would use `def` and `local attribute [instance]` in a section instead?

kbuzzard replied:
Chris pointed out that `instance coe_int_Zmod37 : has_coe ℤ (Zmod37) := ⟨reduce_mod37⟩` is an issue in the wild, because once Zmod37 is a ring there will be a coercion from int anyway. I'll deal with these things soon.

kbuzzard replied:
I have made some troublesome instances local.

kbuzzard committed:
Tidying up this example file.
I've made things local instances, and I've put a bunch of
spaces in following mathlib conventions.

jcommelin commented on docs/tutorial/Zmod37.lean:
```diff
@@ -0,0 +1,211 @@
+  intros a b H,
+  -- H : cond_mod37 a b
+  cases H with l Hl,
+  -- Hk : l * 37 = (b - a)
```
```suggestion
  -- Hl : l * 37 = (b - a)
```

kbuzzard replied:
done

kbuzzard committed:
minor fixups

PatrickMassot committed:
chore(docs/theories/relations): add link

PatrickMassot committed:
Merge branch 'master' into kbuzzard-patch-1
